app/test/unit/unique-coauthors-as-authors-test.ts

```diff
@@ -0,0 +1,94 @@
+import { ITrailer } from '../../src/lib/git'
+import { getUniqueCoauthorsAsAuthors } from '../../src/lib/unique-coauthors-as-authors'
+import { Commit } from '../../src/models/commit'
+import { CommitIdentity } from '../../src/models/commit-identity'
+
+describe('getUniqueCoauthorsAsAuthors', () => {
+  it('can returns empty array for no coauthors', async () => {
+    const trailers: ReadonlyArray<ITrailer> = [
+      { token: 'Signed-Off-By', value: 'test <test@github.com>' },
+    ]
+
+    const commits: Commit[] = [
+      buildTestCommit(trailers),
+      buildTestCommit([]),
+      buildTestCommit(trailers),
+    ]
+
+    const coAuthors = getUniqueCoauthorsAsAuthors(commits)
+    expect(coAuthors).toBeArrayOfSize(0)
+  })
+
+  it('gets coauthor from commit with coauthor', async () => {
+    const email = 'tidy-dev@github.com'
+    const name = 'tidy-dev'
+    const trailers = [buildTestCoAuthorTrailer(email, name)]
+    const commits: Commit[] = [buildTestCommit(trailers)]
+
+    const coAuthors = getUniqueCoauthorsAsAuthors(commits)
+    expect(coAuthors).toBeArrayOfSize(1)
+    const coAuthor = coAuthors[0]
+    expect(coAuthor.email).toBe(email)
+    expect(coAuthor.name).toBe(name)
+  })
+
+  it('does not return duplicate authors', async () => {
+    const email = 'tidy-dev@github.com'
+    const name = 'tidy-dev'
+    const trailers = [buildTestCoAuthorTrailer(email, name)]
+
+    const commits: Commit[] = [
+      buildTestCommit(trailers),
+      buildTestCommit(trailers),
+      buildTestCommit(trailers),
+    ]
+
+    const coAuthors = getUniqueCoauthorsAsAuthors(commits)
+    expect(coAuthors).toBeArrayOfSize(1)
+    const coAuthor = coAuthors[0]
+    expect(coAuthor.email).toBe(email)
+    expect(coAuthor.name).toBe(name)
+  })
+
+  it('can get multiple coauthors on multiple commits', async () => {
+    const first_email = 'tidy-dev@github.com'
+    const first_name = 'tidy-dev'
+    const firstTrailers = [buildTestCoAuthorTrailer(first_email, first_name)]
+
+    const second_email = 'sergiou87@github.com'
+    const second_name = 'Sergio'
+    const secondTrailers = [buildTestCoAuthorTrailer(second_email, second_name)]
```

Comment:
Probably worth adding a case where there are commits with same name but different email (and vice-versa)? Given I _think_ `uniqBy` doesn't work as intended, that would help covering those scenarios.